openwisp_utils/admin.py

```diff
@@ -108,26 +111,41 @@ class ReceiveUrlAdmin(ModelAdmin):
     receive_url_object_arg = 'pk'
     receive_url_name = None
 
+    def add_view(self, request, *args, **kwargs):
+        self.request = request
+        return super().add_view(request, *args, **kwargs)
+
+    def change_view(self, request, *args, **kwargs):
+        self.request = request
+        return super().change_view(request, *args, **kwargs)
+
     def receive_url(self, obj):
         """
         :param obj: Object for which the url is generated
         """
         if self.receive_url_name is None:
             raise ValueError('receive_url_name is not set up')
-        reverse_kwargs = {}
-        if self.receive_url_object_arg:
-            reverse_kwargs = {
-                self.receive_url_object_arg: getattr(obj, self.receive_url_object_arg)
-            }
-        url = reverse(self.receive_url_name, kwargs=reverse_kwargs)
+        if hasattr(settings, "TOPOLOGY_API_URL"):
+            url = settings.TOPOLOGY_API_URL + '/api/receive/'
```

Comment:
ok sorry for not being more explicit.
This is a base module, it must not know anything about any other higher level openwisp module, otherwise we would be creating a mess of interdependency.

The URL that needs to be reversed is configurable as a class attribute right?
So we need to add a new class attribute (eg: `base_url`) defaulting to `None` to `ReceiveUrlAdmin` and use it in its code, that is in django-netjsongraph we must add the base_url class attribute taken from a setting like you do here, I'll try to provide some more examples in the next comments